src/elevateprivilegeshelper.cpp

```diff
@@ -0,0 +1,186 @@
+#include "elevateprivilegeshelper.h"
+
+#include <QDebug>
+#include <QFile>
+#include <QSocketNotifier>
+#include <QStandardPaths>
+#include <QUuid>
+
+#include <fcntl.h>
+#include <signal.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <vector>
+
+InitiallyStoppedProcess::~InitiallyStoppedProcess()
+{
+    kill();
+}
+
+bool InitiallyStoppedProcess::reset(const QString& exePath, const QStringList& exeOptions,
+                                    const QString& workingDirectory)
+{
+    kill();
+
+    // convert arguments and working dir into what the C API needs
+
+    std::vector<QByteArray> args;
+    args.reserve(exeOptions.size() + 1);
+    args.emplace_back(exePath.toLocal8Bit());
+    for (const auto& opt : exeOptions)
+        args.emplace_back(opt.toLocal8Bit());
+    const auto wd = workingDirectory.toLocal8Bit();
+
+    std::vector<char*> argsArray(args.size() + 1);
+    for (size_t i = 0; i < args.size(); ++i)
+        argsArray[i] = args[i].data();
+    argsArray.back() = nullptr;
+
+    // fork
+    m_pid = fork();
```

Comment:
forking means we COW the whole address space which can be really slow when that is large. that would e.g. be the case when you start hotspot, analyze some large file and then rerun the record from there.

is there a way this could use QProcess such that we can leverage vfork/posix_spawn when available? or is there maybe something like that in boost which we could use to get this behavior in a more efficient manner?

see also: https://codereview.qt-project.org/c/qt/qtbase/+/417829

Reply:
sure this could be improved. Don't know what's available in boost for that. vfork seems a bit dangerous to me, for posix_spawn I suppose one would need to move this initially-stopped-exec functionality in a separate helper binary or something.

Reply:
we can only use vfork if we create a new binary, since vfork only allows for `__exit()` and `exec*` calls. I think `clone` with `CLONE_WM | CLONE_VFORK` might be the easiest way to solve this.

Reply:
From my understanding, VFORK suspends the calling process until exec is called, which we do not want here at all. I don't think VFORK is usable for this. posix_spawn and a helper process should work though.